lisa/sut_orchestrator/azure/features.py

```diff
@@ -783,3 +790,126 @@ def get_azure_disk_type(disk_type: schema.DiskType) -> str:
+        available_sizes = vm_operations.list_available_sizes(
+            node_context.resource_group_name, node_context.vm_name
+        )
+        vm_sizes_with_capability = platform.get_eligible_vm_sizes(
+            node_runbook.location, self._log
+        )  # add location in node_context
+
+        current_vm_size = next(
+            (x for x in vm_sizes_with_capability if x.vm_size == node_runbook.vm_size),
+            None,
+        )
+        assert current_vm_size, "cannot find current vm size in eligible list"
+
+        resizable_vmsizes: List[AzureCapability] = []
+        for size in available_sizes:
+            """sizes.append(size.as_dict())
+            print("avail siz: ", size.as_dict())
+            print("avail siz: ", size.serialize())"""
+
+            vm_size_name = size.as_dict()["name"]
+            # get eligible vm sizes and its capability data.
+            new_vm_size = next(
+                (x for x in vm_sizes_with_capability if x.vm_size == vm_size_name), None
+            )
+            if not new_vm_size:
+                continue
+
+            resizable_vmsizes.append(new_vm_size)
+
+        vmsize_names: List[str] = []
+        # loop to check if the vm size meeting the action
+        for candidate_size in resizable_vmsizes[:]:
+            # The commented out code ensures that none isn't returned. Is it necessary or should I erase it?
+            """current_network_interface = current_vm_size.capability.network_interface
+            if not isinstance(
+                current_network_interface, schema.NetworkInterfaceOptionSettings
+            ):
+                raise LisaException(
+                    "current_network_interface is not of type "
+                    "NetworkInterfaceOptionSettings."
+                )
+            current_data_path = current_network_interface.data_path"""
+            current_data_path = current_vm_size.capability.network_interface.data_path
+            """candidate_network_interface = candidate_size.capability.network_interface
+            if not isinstance(
+                candidate_network_interface, schema.NetworkInterfaceOptionSettings
+            ):
+                raise LisaException(
+                    "candidate_network_interface is not of type "
+                    "NetworkInterfaceOptionSettings."
+                )
+            candidate_data_path = candidate_network_interface.data_path"""
+            candidate_data_path = candidate_size.capability.network_interface.data_path
+
+            if (
+                schema.NetworkDataPath.Sriov in current_data_path
+                and schema.NetworkDataPath.Sriov not in candidate_data_path
+            ):
+                continue
+
+            # Below commented lines are safety checks but the code works without them. Should they be kept?
+            candidate_count = candidate_size.capability.core_count
+            """if not isinstance(candidate_count, int):
```

Comment:
You can change it to assert, it's simpler.

Reply:
done